### DISCUSSION TITLE
define one spotlight emitter for all spotlights

### DISCUSSION THREAD

**Discussion by yangocean-sudo** [mitsuba-renderer/mitsuba3#570] [Q&A] [answered]
Hi, I want to emit a lot of spotlight in the scene. One possible way is to make each spotlight individual, like this:
```   
<emitter type="spot" id="spotlight1">
        <transform name="to_world">
       		<lookat origin="0,-1,0" target="3.1, 0, 0" up="0, 0, 0"/>
   	</transform>
        <rgb name="intensity" value="20000.0"/>
        <float name="cutoff_angle" value="0.1"/>
</emitter>
<emitter type="spot" id="spotlight2">
        <transform name="to_world">
       		<lookat origin="0,-2,0" target="2.9, 0, 0" up="0, 0, 0"/>
   	</transform>
        <rgb name="intensity" value="20000.0"/>
        <float name="cutoff_angle" value="0.1"/>
</emitter>
```
But the amount of emitters is very large and the compilation time will be very long, is there a method to define one spotlight emitter for all spotlights, like:
```
<emitter type="spot" id="spotlight">
        <transform name="to_world">
       		<lookat origin="0,-2,0" target="2.9, 0, 0" up="0, 0, 0"/>
       		<lookat origin="0,-1,0" target="3.1, 0, 0" up="0, 0, 0"/>
       		<lookat origin="0,-3,0" target="3.0, 0, 0" up="0, 0, 0"/>
   	</transform>
        <rgb name="intensity" value="20000.0"/>
        <float name="cutoff_angle" value="0.1"/>
</emitter>
```
Thank you.

**Comment by njroussel** [comment, marked as answer]
Hi @yangocean-sudo 

If you want to understand why this is happening, take a look at this discussion: https://github.com/mitsuba-renderer/mitsuba3/discussions/565

I've just pushed a change to `master` which should resolve the long compilation times with multiple emitters: https://github.com/mitsuba-renderer/mitsuba3/pull/571. You'll need to compile the project yourself if you want this new feature or wait until we push out a new release to PyPI.
